### PR TITLE
Introduced support for revoking tokens

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**

### CODE REVIEW

File: Sources/OktaIdx/IDXClient.swift [hunk 1]
```diff
@@ -311,6 +318,33 @@ public final class IDXClient: NSObject {
 
         self.api.client = self
     }
+    
+    @objc(revokeToken:type:completion:)
+    public func revoke(token: Token, type: Token.RevokeType, completion: @escaping(_ successful: Bool, _ error: Error?) -> Void) {
+        let selectedToken: String?
```
Comment (oleggnidets-okta) [inline Sources/OktaIdx/IDXClient.swift:324]: You may avoid optional. Just make `let selectedToken: String`.

File: Sources/OktaIdx/Extensions/IDXClient+Response.swift [hunk 2]
```diff
@@ -184,6 +184,11 @@ public extension IDXClient {
         
         /// The type of this token.
         @objc public let tokenType: String
+        
+        @objc public enum RevokeType: Int {
+            case refreshToken
+            case accessAndRefreshToken
```
Comment (oleggnidets-okta) [inline Sources/OktaIdx/Extensions/IDXClient+Response.swift:190]: What about ID token?

**Review by oleggnidets-okta — APPROVED** [review]

File: Sources/OktaIdx/IDXClient.swift (hunk 1 above)
Comment (mikenachbaur-okta) [reply Sources/OktaIdx/IDXClient.swift:324]: The `refreshToken` property may be nil, so this variable still needs to remain optional.

File: Sources/OktaIdx/Extensions/IDXClient+Response.swift (hunk 2 above)
Comment (mikenachbaur-okta) [reply Sources/OktaIdx/Extensions/IDXClient+Response.swift:190]: The [revoke API only provides these two options](https://developer.okta.com/docs/guides/revoke-tokens/revokeatrt/), from what I can tell.

**Pull request merged**